spec/support/synchronization_helpers.rb

```diff
@@ -1,4 +1,25 @@
 module SynchronizationHelpers
+  def expect_in_fork
+    read, write = IO.pipe
+
+    pid = fork do
+      read.close
+
+      yield
+
+      Marshal.dump(true, write)
+    end
+
+    # Wait for fork to finish, retrieve its output
+    write.close
+    result = read.read
+    Process.wait(pid)
+
+    # Expect fork and assertions to have completed successfully.
+    # rubocop:disable Security/MarshalLoad
```

Comment:
Is there a better way to do this?

Reply:
Could use the status code of the child process here? https://ruby-doc.org/core-2.6.1/Process.html#method-c-waitpid

The status code is available after `wait` in the global variable `$?`.

Reply:
Oh yeah, a status code might be appropriate because we don't actually need what the fork sends us, just that it didn't fail. I'll look into that.

Reply:
I think this worked well. Marking as resolved.